src/main/drivers/flash.c

```diff
@@ -259,4 +274,90 @@ const flashGeometry_t *flashGetGeometry(void)
 
     return &noFlashGeometry;
 }
+
+/*
+ * Flash partitioning
+ *
+ * Partition table is not currently stored on the flash, in-memory only.
+ *
+ * Partitions are required so that Badblock management (inc spare blocks), FlashFS (Blackbox Logging), Configuration and Firmware can be kept separate and tracked.
+ *
+ * Currently, to keep things simple (and working), the following rules apply:
+ *
+ * 1) order of partitions in the paritions table strictly defined as follows
```

Comment:
I think we are overspecifying here. If the order / absolute positioning is strictly enforced, then we do not need a `usage` flag. But I think it would be more attractive to not enforce absolute positioning, and hence force consumers of flash resources to use `flashFindPartitionByUsage()` to get the partition they want. If we don't, then sooner or later we are going to be in the situation where we have a mix of `flashFindPartitionByUsage()` and absolute reference into the array to get the appropriate partition.

Also, 'parTitions'.